spinalcordtoolbox/scripts/sct_compute_snr.py

```diff
@@ -149,7 +149,7 @@ def main(argv=None):
     # Check parameters
     if method in ['diff', 'single']:
         if not fname_mask:
-            raise parser.error(f"Argument '-m' must be specified when using '-method {method}'.")
+            parser.error(f"Argument '-m' must be specified when using '-method {method}'.")
```

Comment:
This makes sense to me!

> [`raise`](https://docs.python.org/3/tutorial/errors.html#raising-exceptions)
>
> The [raise](https://docs.python.org/3/reference/simple_stmts.html#raise) statement allows the programmer to force a specified exception to occur. The sole argument to [raise](https://docs.python.org/3/reference/simple_stmts.html#raise) indicates the exception to be raised. **This must be either an exception instance or an exception class (a class that derives from [Exception](https://docs.python.org/3/library/exceptions.html#Exception)).**

> [`ArgumentParser.error(message)`](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.error)
>
> This method prints a usage message including the message to the standard error and terminates the program with a status code of 2.

Just for fun, I tried using `raise` with a dummy function and got this:

```
Traceback (most recent call last):
  File "/home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/lib/python3.7/code.py", line 90, in runcode
    exec(code, self.locals)
  File "<input>", line 1, in <module>
TypeError: exceptions must derive from BaseException
```

I was curious why `raise parser.error` doesn't throw the same error, but then I realized... `raise parser.error` _doesn't even get to `raise`_; the program `exit`s before the `raise` statement can be invoked. :facepalm: 

Definitely unnecessary, then. :slightly_smiling_face:

Reply:
Exactly!

And FYI, you can get the same result with `raise None`, because functions that don't return anything explicitly (such as your dummy function) actually return `None`.

(Anecdotally, I'm guilty of writing the opposite bug at least once: creating an Exception object without raising it. Python will happily run the statement, create the exception object, then discard it, as if discarding an unused return value of a function call.)